src/Catel.Core/Catel.Core.Shared/Logging/LogListenerBase.cs

```diff
@@ -164,9 +165,9 @@ public TimeDisplay TimeDisplay
         /// <param name="time">The time.</param>
         void ILogListener.Write(ILog log, string message, LogEvent logEvent, object extraData, LogData logData, DateTime time)
         {
-            // We always want the Argument class logging, no matter what
-            var isArgumentLog = log.TargetType == ArgumentType;
-            if (!isArgumentLog)
+            // If the log is a catel log and the AlwaysLog flag is set, log the message.
+            var catelLog = log as ICatelLog;
+            if (!catelLog?.AlwaysLog ?? true)
```

Comment:
You are right, I just misread it. Maybe we should add unit tests so we can prove this works ;-)